### ISSUE TITLE
New Big Pack Bloc Stack Overflow issue

### ISSUE THREAD

**Issue by nikitka5702** [Gorniv/vscode-flutter-files#11]
Flutter version:
```
$ flutter --version
Flutter 1.8.3 • channel beta • https://github.com/flutter/flutter.git
Framework • revision e4ebcdf6f4 (5 weeks ago) • 2019-07-27 11:48:24 -0700
Engine • revision 38ac5f30a7
Tools • Dart 2.5.0 (build 2.5.0-dev.1.0 0ca1582afd)
```
Seems like on latest dart version it is not possible to create class with recursive field (current implementation in repository template, which was introduced in fix for #10 )
I've used `New Big Pack Bloc` to create home bloc.
main.dart file
```dart
import 'package:flutter/material.dart';

import 'home/index.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: HomePage(),
    );
  }
}
```
Debug log
```
I/flutter ( 9295): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter ( 9295): The following StackOverflowError was thrown building MediaQuery(MediaQueryData(size: Size(411.4,
I/flutter ( 9295): 683.4), devicePixelRatio: 2.6, textScaleFactor: 1.0, platformBrightness: Brightness.light, padding:
I/flutter ( 9295): EdgeInsets.zero, viewPadding: EdgeInsets.zero, viewInsets: EdgeInsets.zero, alwaysUse24HourFormat:
I/flutter ( 9295): false, accessibleNavigation: false, disableAnimations: false, invertColors: false, boldText:
I/flutter ( 9295): false)):
I/flutter ( 9295): Stack Overflow
I/flutter ( 9295):
I/flutter ( 9295): User-created ancestor of the error-causing widget was:
I/flutter ( 9295):   Scaffold 
lib\home\home_page.dart:10
I/flutter ( 9295):
I/flutter ( 9295): When the exception was thrown, this was the stack:
I/flutter ( 9295): #0      new HomeRepository (package:sample_app/home/home_repository.dart)
I/flutter ( 9295): #1      new HomeRepository 
package:sample_app/home/home_repository.dart:4
I/flutter ( 9295): #2      new HomeRepository 
package:sample_app/home/home_repository.dart:4
I/flutter ( 9295): #3      new HomeRepository 
package:sample_app/home/home_repository.dart:4
```

**Comment by spiridonovmv** [comment]
Problem in repository class. You need rename:

```
class YourRepository {
  final YourProvider _YourProvider = new YourProvider();

...
}
```

**Comment by Gorniv** [comment]
fixed at 1.5.3
